maptivist/src/main/webapp/script.js

```diff
@@ -248,26 +244,47 @@ var GoogleAuth;
     }
   }
 
+  function handleAuthClick2() {
+    //opens map if signed in, prompts to sign in if not signed in
+    if (GoogleAuth.isSignedIn.get()) {
+
+      openForm();
+    } else {
+
+      GoogleAuth.signIn();
+    }
+  }
+
+
   function revokeAccess() {
     GoogleAuth.disconnect();
   }
 
   function setSigninStatus(isSignedIn) {
     var user = GoogleAuth.currentUser.get();
     var isAuthorized = user.hasGrantedScopes(SCOPE);
+    $('#sign-in-or-out-button').click(function() {
+        handleAuthClick();
+      });
+      $('#myButton').click(function() {
+        handleAuthClick2();
+      });
+      $('#authLink').click(function() {
+        handleAuthClick();
+      });
```

Comment:
Now that I re-read the code, I think there might be even a simpler way to do this, and help it be readable.

More informative function names will help the reader understand their purpose.
handleAuthClick -> signInOrOut
handleAuthClick2 ->signInOrOpenMap

Also, I overlooked this before (sorry) and perhaps I was confused by the control flow, but the callbacks don't actually change. They just check the isSignedIn status. In that case, they never need to change. In that case, maybe they can just be put back where they were, or perhaps just put them in the HTML as `onclick` handlers?